consensus/src/testing/mock.rs

```diff
@@ -494,6 +494,98 @@ impl DataProvider {
     }
 }
 
+type SavedUnit = Vec<u8>;
+
+struct UnitSaver {
+    rx: UnboundedReceiver<SavedUnit>,
+    units: Arc<Mutex<Vec<SavedUnit>>>,
+}
+
+impl UnitSaver {
+    async fn run(mut self) {
```

Comment:
Personally, I would feel more comfortable with these interfaces being synchronous. It is crucial that we don't continue with the unit before it's stored and we don't start creating the new ones until we've loaded all from the disk on startup.

If we're worried about the performance then we're really saying that persisting X bytes on disk might take longer than sending them over the wire (what we will be sending over the wire will be more in practice b/c of all the message envelopes, etc.).

Reply:
But it is a mock for tests. It has nothing in common with actual implementation of this saver. I'm not sure if this is the place where you wanted to add this comment.

Reply:
Ah yes, wrong place but the idea is the same.

Yes, I read the docs, it's just that making this interface non-async makes it more explicit that we have to wait. Even if you called `save.await` we don't gain anything from making this async as we want to always wait anyway.